src/libraries/Common/src/System/Net/Http/aspnetcore/Http2/Hpack/Huffman.cs

```diff
@@ -307,128 +402,148 @@ public static (uint encoded, int bitLength) Encode(int data)
         /// <returns>The number of decoded symbols.</returns>
         public static int Decode(ReadOnlySpan<byte> src, ref byte[] dstArray)
         {
+            // The code below implements the decoding logic for a HPack huffman encoded literal values.
+            // https://httpwg.org/specs/rfc7541.html#string.literal.representation
+            //
+            // To decode a symbol, we traverse the decoding lookup table tree by 8 bits for each lookup
+            // until we found a leaf - which contains decoded symbol (octet)
+            //
+            // see comments in GenerateDecodingLookupTree() describing decoding table
+
             Span<byte> dst = dstArray;
             Debug.Assert(dst != null && dst.Length > 0);
 
+            if (s_decodingTree == null)
+            {
+                Volatile.Write(ref s_decodingTree, GenerateDecodingLookupTree());
```

Comment:
The Volatile.Write only guarantees you've flushed the result to memory, it doesn't guarantee anybody reads it, you'd need a Volatile.Read for that which you don't want happening on every request. You could end up running this initialization code on every core at startup. What's the cost?

Reply:
I have discussed this tradeoff with Cory. I am aware about possibility of multiple initialization with current implementation.

Bellow is a benchmark.

BenchmarkDotNet=v0.12.1, OS=Windows 10.0.19042
AMD Ryzen 5 3600, 1 CPU, 12 logical and 6 physical cores
.NET Core SDK=6.0.100-alpha.1.20472.11
  [Host]     : .NET Core 6.0.0 (CoreCLR 6.0.20.47901, CoreFX 6.0.20.47901), X64 RyuJIT
  Job-LKQRKP : .NET Core 6.0.0 (CoreCLR 6.0.20.46807, CoreFX 6.0.20.46807), X64 RyuJIT

Server=True  Toolchain=.NET Core 5.0  RunStrategy=Throughput

|               Method |     Mean |     Error |    StdDev |      Op/s |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|--------------------- |---------:|----------:|----------:|----------:|-------:|------:|------:|----------:|
| GenerateDecodingTree | 5.296 us | 0.0441 us | 0.0391 us | 188,805.9 | 0.0687 |     - |     - |   7.52 KB |

Please let me know if 5us is acceptable or if you recommend some further optimization in this area.

Reply:
Ok, that's pretty cheep. Thanks for checking.

Reply:
> you'd need a Volatile.Read for that which you don't want happening on every request

Technically you need a volatile read anyway.